### PR TITLE
Reduce VAD latency.

### PR DESCRIPTION
**Pull request by hksfang**
Utilize whisper transcription for speech interim chunks to achieve transcribing while speaking.
Test video: https://drive.google.com/file/d/19I73GiIcz3Rkj6zb2KuTD93GLXFFd3y8/view?usp=sharing

### CODE REVIEW

**Comment by Shaunwei** [comment]
Thanks for the PR. @hksfang with and w/o the change, qq, how much latency have it reduced?

File: realtime_ai_character/websocket_routes.py [hunk 1]
```diff
@@ -348,8 +366,20 @@ async def stop_audio():
             # handle binary message(audio)
             elif 'bytes' in data:
                 binary_data = data['bytes']
+                # 0. Handle interim speech.
+                if speech_recognition_interim:
+                    interim_transcript: str = (await asyncio.to_thread(speech_to_text.transcribe,
+                        binary_data, platform=platform,prompt=current_speech, suppress_tokens=[0, 11, 13, 30])).strip()
```
Comment (Shaunwei) [inline realtime_ai_character/websocket_routes.py:372]: qq, what does suppress_tokens do?

Comment (hksfang) [reply realtime_ai_character/websocket_routes.py:372]: It's for suppressing punctuation marks`!?.,`, I did some test and found out if whisper doesn't output punctuation, the interim transcript prompt works better.

The numbers come from the output of whisper tokenizer, via [this issue](https://github.com/openai/whisper/discussions/589)

**Comment by hksfang** [comment]
> Thanks for the PR. @hksfang with and w/o the change, qq, how much latency have it reduced?

The result varies on the speaker's speaking pattern, for longer speech with more 'gaps' in between, this PR reduces considerable latency, but for shorter speech or speech with little to no gaps, this PR doesn't do much.

I found it difficult to benchmark this change, but I can provide an example.
Using this [audio file](https://drive.google.com/file/d/1JTS8nMhnurrqsHVKlq8tcJ_k4uBoC8zH/view?usp=drive_link) as example, before this PR, transcription after the end of speech took 5.5s, after this PR, only 2s, that's around 54% reduction in transcription latency.

**Comment by Shaunwei** [comment]
> > Thanks for the PR. @hksfang with and w/o the change, qq, how much latency have it reduced?
> 
> The result varies on the speaker's speaking pattern, for longer speech with more 'gaps' in between, this PR reduces considerable latency, but for shorter speech or speech with little to no gaps, this PR doesn't do much.
> 
> I found it difficult to benchmark this change, but I can provide an example. Using this [audio file](https://drive.google.com/file/d/1JTS8nMhnurrqsHVKlq8tcJ_k4uBoC8zH/view?usp=drive_link) as example, before this PR, transcription after the end of speech took 5.5s, after this PR, only 2s, that's around 54% reduction in transcription latency.

Wow. this is super awesome. Thanks for making the change. LGTMed

**Pull request merged**